src/App.tsx

```diff
@@ -1,66 +1,59 @@
-import { useEffect, useState } from "react";
+import { useEffect } from "react";
 import TodoForm from "./components/todo/TodoForm";
 import TodoList from "./components/todo/TodoList";
-import { UserModal } from "./models/user";
 import Login from "./components/shared/Login";
-import { useSelector } from "react-redux";
-import { gapi } from "gapi-script";
-import { todoList } from "./store/todoReducer";
-
-const clientId = process.env.REACT_APP_CLIENTID;
+import { useDispatch, useSelector } from "react-redux";
+import { reducerSate } from "./store/todoReducer";
+import Button from "./components/shared/form/Button";
+import { loginTodoAction } from "./store";
 
 function App() {
-  const state = useSelector((state: todoList) => state);
-  const [isLogedin, setIsLogedin] = useState({});
+  const state = useSelector((state: reducerSate) => state);
+  const dispatch = useDispatch();
   var accessToken = window.localStorage.getItem("accessToken");
-  useEffect(() => {
-    function start() {
-      gapi.client.init({
-        clientId: clientId,
-        scope: "",
-      });
-    }
-    gapi.load("client:auth2", start);
-  });
+  const userName = window.localStorage.getItem("userName")
+    ? window.localStorage.getItem("userName")
+    : "";
+  const setIsLogedin = (bolval: boolean) => {
+    dispatch(loginTodoAction(bolval));
```

Comment:
We can manipulate the auth related state in login page itself. What do you think?

Reply:
Kindly provide more detail